File: pmcid_test.ts
//
// Test our PMCID methods
//
import { assertEquals, assertNotEquals } from "@std/assert";

import { normalizePMCID, validatePMCID } from "./pmcid.ts";
import { verifyPMCID } from "./pmcid_record.ts";

Deno.test("test pmcid", async () => {
  const varified_ids: string[] = [
    "PMC11021482",
    "10557066",
  ];

  for (const id of varified_ids) {
    // Normalize
    const normalized = normalizePMCID(id);
    assertNotEquals(normalized, undefined);
    console.log(`Normalized PMCID: ${normalized}`);
    // Validate
    assertEquals(validatePMCID(id), true);
    // Verify is via pmcid.org
    const isOK: boolean = await verifyPMCID(id);
    assertEquals(isOK, true);
  }
});

File: pmcid_test.go
package metadatatools

import "testing"

func TestPMCID(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping network test in short mode")
	}

	verifiedIDs := []string{
		"PMC11021482",
		"10557066",
	}

	for _, id := range verifiedIDs {
		normalized := NormalizePMCID(id)
		if normalized == "" {
			t.Errorf("NormalizePMCID(%q) returned an empty string", id)
		}
		t.Logf("Normalized PMCID: %s", normalized)

		if !ValidatePMCID(id) {
			t.Errorf("ValidatePMCID(%q) = false, expected true", id)
		}

		if !VerifyPMCID(id) {
			t.Errorf("VerifyPMCID(%q) = false, expected true", id)
		}
	}
}
